test/functional-api/integration/callouts/close-state-callouts.it-spec.ts

```diff
@@ -0,0 +1,655 @@
+        ${TestUser.QA_USER}
+        ${TestUser.NON_HUB_MEMBER}
+      `(
+        `User: "$userRole" get errorwhen send message to closed callout with id: ${opportunityCalloutId}`,
+        async ({ userRole }) => {
+          // Act
+          const res = await postCommentInCallout(
+            opportunityCalloutId,
+            'comment on discussion callout',
+            userRole
+          );
+
+          // Assert
+          expect(res.text).toContain(
+            `"New collaborations to a closed Callout with id: '${opportunityCalloutId}' are not allowed!"`
+          );
+        }
+      );
+    });
+  });
+});
+
+// The suite contains scenarios for 'card create' and 'card comment'. Card Update / Delete to be added on later stage (low priority)
+describe('Callout - Close State - User Privileges Cards', () => {
+  let hubCalloutId = '';
+  let challengeCalloutId = '';
+  let opportunityCalloutId = '';
+  const cardCalloutNameId = 'card-default';
+  let hubAspectId = '';
+  let aspectCommentsIdChallenge = '';
+  cardNameID = `aspect-name-id-${uniqueId}`;
+
+  beforeAll(async () => {
+    const preconditions = async (calloutId: string) => {
+      const resAspectonHub = await createAspectOnCallout(
+        calloutId,
+        'aspectDisplayName',
+        cardNameID
+      );
+      const aspectDataCreate = resAspectonHub.body.data.createAspectOnCallout;
+      const aspectId = aspectDataCreate.id;
+      aspectCommentsIdChallenge = aspectDataCreate.comments.id;
+
+      const messageRes = await mutation(
+        sendComment,
+        sendCommentVariablesData(
+          aspectCommentsIdChallenge,
+          'test message on aspect'
+        ),
+        TestUser.HUB_MEMBER
+      );
+      msessageId = messageRes.body.data.sendComment.id;
+      hubAspectId = aspectDataCreate.id;
+      await updateCallout(
+        calloutId,
+        calloutDisplayName,
+        cardCalloutNameId,
+        'calloutDescription update',
+        CalloutState.CLOSED,
+        CalloutType.COMMENTS
+      );
+    };
+
+    const hubCallout = await getHubCalloutByNameId(
+      entitiesId.hubId,
+      cardCalloutNameId
+    );
+    hubCalloutId = hubCallout[0].id;
+    await preconditions(hubCalloutId);
+
+    const challengeCallout = await getChallengeCalloutByNameId(
+      entitiesId.hubId,
+      entitiesId.challengeId,
+      cardCalloutNameId
+    );
+    challengeCalloutId = challengeCallout[0].id;
+    await preconditions(challengeCalloutId);
+
+    const opportunityCallout = await getOpportunityCalloutByNameId(
+      entitiesId.hubId,
+      entitiesId.opportunityId,
+      cardCalloutNameId
+    );
+    opportunityCalloutId = opportunityCallout[0].id;
+    await preconditions(opportunityCalloutId);
+  });
+
+  afterAll(async () => {
+    await deleteCallout(opportunityCalloutId);
+    await deleteCallout(challengeCalloutId);
+    await deleteCallout(hubCalloutId);
+  });
+
+  afterEach(async () => {
+    await delay(3000);
```

Comment:
put a comment why is this required

Reply:
If the question is related to the `delay()` function, it is used in the tests where we send / delete comments, because, sometimes there are delays, in the response.
Regarding the `afterEach()` block - no in the particular case it is not required and will be removed with the next commit.